static/js/app.ts

```diff
@@ -286,10 +291,29 @@ function attachMainEditorEvents(editor: HedyEditor) {
     askPromptOpen = false;
     $('#runit').css('background-color', '');
     theGlobalEditor.clearErrors();
+    theGlobalEditor.clearIncorrectLines();
     //removing the debugging state when loading in the editor
     stopDebug();
   });
 
+  editor.on('click', (event: MouseEvent) => {
+    if (theGlobalEditor instanceof HedyCodeMirrorEditor) {
+      if (!theGlobalEditor.hasIncorrectLinesDecorations()) return;
+      const pos = theGlobalEditor.getPosFromCoord(event.x, event.y);
+      if (pos == null) return;
+      const index = theGlobalEditor.indexOfErrorInPos(pos)
+      if (index == null) {
+        // Hide error, warning or okbox
+        error.hide();
+      } else {
+        // Show error for this line
+        let mapError = theGlobalSourcemap[index];
+        error.hide();
+        error.show(ClientMessages['Transpile_error'], mapError.error);
+      }
+    }
+  })
```

Comment:
Couldn't this whole block be a feature of `HedyCodeMirrorEditor` ?

```ts
new HedyCodeMirrorEditor({
  // Given an error manager, the editor can handle it?
  errorManager: error,
});
```

Or is the problem you need access to the source map?

Reply:
Nice, idea!